### PR TITLE
feat: adapted to new ipfs-repo API

### PR DESCRIPTION
**Pull request by pgte**
ipfs-repo dependents are:
- [x] js-ipfs, this PR.
- [x] ipfs-block-service - [PR](https://github.com/ipfs/js-ipfs-block-service/pull/66)
- [x] ipld-resolver - [PR](https://github.com/ipld/js-ipld-resolver/pull/84)
- [x] ipfs-bitswap - [PR](https://github.com/ipfs/js-ipfs-bitswap/pull/144)

### CODE REVIEW

**Comment by pgte** [comment]
Depends on https://github.com/ipfs/js-ipfs-repo/pull/140

**Comment by diasdavid** [comment]
@pgte please run `npm run test:interop` too

**Comment by pgte** [comment]
@diasdavid tested and it runs locally, but first you need to link these to the respective branches:

```
$ npm link ipfs-repo ipfs-block-service ipld-resolver ipfs-bitswap
```

File: src/core/components/config.js [hunk 1]
```diff
@@ -15,23 +12,9 @@ module.exports = function config (self) {
 
       if (!key) {
         return self._repo.config.get(callback)
+      } else {
+        return self._repo.config.get(key, callback)
```
Comment (diasdavid) [inline src/core/components/config.js]: Doesn't the IPFS repo figure this out by itself now?

File: src/core/components/config.js [hunk 2]
```diff
@@ -42,13 +25,7 @@ module.exports = function config (self) {
         return callback(new Error('Invalid value type'))
       }
```
Comment (diasdavid) [inline src/core/components/config.js]: Isn't this validation happening inside the repo itself?

Comment (pgte) [inline src/core/components/config.js]: It should. https://github.com/ipfs/js-ipfs-repo/pull/140 supports this now (via https://github.com/ipfs/js-ipfs-repo/commit/8e8055ddae3f4f6b44ebb4f8932393dc95327039).
Fixed.

**Comment by diasdavid** [comment]
@pgte tests fail because of:

```
/Users/koruza/code/js-ipfs/src/core/components/start.js:51
      self._blockService.goOnline(self._bitswap)
                         ^

TypeError: self._blockService.goOnline is not a function
    at series (/Users/koruza/code/js-ipfs/src/core/components/start.j
```

Wasn't this change supposed to be contemplated in the awesome endeavor?

**Comment by pgte** [comment]
@diasdavid I hadn't included this change in the awesome endeavour, it was in a separate PR which wasn't listed here, I didn't think you would merge it as part of this.
Anyway, thanks for making this change!

**Pull request merged**